megaavr/cores/dxcore/wiring_analog.c

```diff
@@ -890,40 +915,39 @@ void analogReadEnableDifferential(uint8_t negPin) {
 // Designed expecting user to pass in number of results, not number set to the register
 void analogReadSampleNum(uint8_t numSamples) {
   while (ADC0.COMMAND & ADC_STCONV_bm) {}
-    ADC0.CTRLB &= 0b11111000;   //Setting accumulation to 0
+  ADC0.CTRLB &= 0b11111000;   //Setting accumulation to 0
 
-    switch (numSamples) {
-    case 0:
-      //Empty case so that 0 doesn't through an error
-      break;
-    case 1:
-      //Does nothing since 1 result accumulated is no accumulation
-      break;
-    case 2:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC2_gc;
-      break;
-    case 4:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC4_gc;
-      break;
-    case 8:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC8_gc;
-      break;
-    case 16:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC16_gc;
-      break;
-    case 32:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC32_gc;
-      break;
-    case 64:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC64_gc;
-      break;
-    case 128:
-      ADC0.CTRLB |= ADC_SAMPNUM_t::ADC_SAMPNUM_ACC128_gc;
-      break;
-    default:
-      badArg("Unexpected value");
-      break;
-    }
+  switch (numSamples) {
+  case 0:
+    //Empty case so that 0 doesn't through an error
+    break;
+  case 1:
+    //Does nothing since 1 result accumulated is no accumulation
+    break;
+  case 2:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC2_gc;
+    break;
+  case 4:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC4_gc;
+    break;
+  case 8:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC8_gc;
+    break;
+  case 16:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC16_gc;
+    break;
+  case 32:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC32_gc;
+    break;
+  case 64:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC64_gc;
+    break;
+  case 128:
+    ADC0.CTRLB |= ADC_SAMPNUM_ACC128_gc;
+    break;
+  default:
```

Comment:
We might want to add back in some kind of error print statement to the default, I think I had originally used
`fprintf(stderr, "%s", "Unexpected Parameter")`

Reply:
Not sure the best way to handle this. I'm not sure what `fprintf` will do on Arduino (or if it's available), we could do a Serial print but I'm not sure how useful that is. Another alternative we could consider is changing the parameter to this function to be an enum type so that only enum members can be passed and input validation occurs at compile time.

Reply:
Gotcha, I wasn't sure if maybe the print statement would come out in console during compilation but ye youre prob right I dont think it would work, maybe the enum could work or we just add some documentation telling people to only use certain values